src/standard/notify-path.html

```diff
@@ -146,29 +157,56 @@
         var array;
         var last = parts[parts.length-1];
         if (parts.length > 1) {
+          // Loop over path parts[0..n-2] and dereference
           for (var i=0; i<parts.length-1; i++) {
             var part = parts[i];
-            prop = prop[part];
-            if (array && (parseInt(part) == part)) {
-              parts[i] = Polymer.Collection.get(array).getKey(prop);
+            if (array && part[0] == '#') {
+              // Part was key; lookup item in collection
+              prop = Polymer.Collection.get(array).getItem(part);
+            } else {
+              // Get item from simple property dereference
+              prop = prop[part];
+              if (array && (parseInt(part) == part)) {

```

Comment:
We might want to specify the radix here, to eliminate an entire class of odd behavior 

![screen shot 2015-10-08 at 5 47 35 pm](https://cloud.githubusercontent.com/assets/883126/10383511/bc2ecf20-6de4-11e5-8e7d-6bbf118dfa48.png)